src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -4029,11 +3987,15 @@ void MarkFieldsOnTypeHierarchy (ref ReflectionPatternContext reflectionContext,
 
 						var methodCalling = reflectionContext.MethodCalling;
 						reflectionContext.RecordRecognizedPattern (field, () => _markStep.MarkField (field, new DependencyInfo (DependencyKind.AccessedViaReflection, methodCalling)));
+						foundMatch = true;
 					}
 
 					type = type.BaseType?.Resolve ();
 					onBaseType = true;
 				}
+
+				if (!foundMatch)
+					reflectionContext.RecordUnrecognizedPattern ($"Reflection call '{reflectionContext.MethodCalled.FullName}' inside '{reflectionContext.MethodCalling.FullName}' could not resolve any field on type `{receivedType}`.");
```

Comment:
This is problematic for two reasons:
* For the cases where we don't filter on name at all, we don't want to fail if there's nothing - that is the case of `GetFields` - it's not a problem calling that on a type without fields - so we should not warn.
* As we discussed we should probably not warn if we find `GetField("fieldName")` and can't find `fieldName` on the type. We did recognize the pattern, but it didn't end up marking anything. I know that's how it worked before, but I think we should change that.

/cc @MichalStrehovsky - for another view on this.

Reply:
Should we then remove all calls to `reflectionContext.AnalyzingPattern` inside `HandleCall` and only call it inside `RequireDynamicallyAccessedMembers`?

Reply:
Maybe - but I would not worry about that. We will have to change the `reflectionContext` anyway. It doesn't have enough capability currently to record events which are not about methods (for example storing a value in a field can't be expressed in it now).

The other change which will have to happen is to how the `AnalyzingPattern` works. Currently the context is supposed to guarantee that if we call `AnalyzingPattern` then we will call either "recognized" or "unrecognized" methods before returning. But this is already broken by the generic `RecordHandledPattern` which is basically "Don't worry - I did the right thing", but it actually doesn't report either "recognized" or "unrecognized" (it reports nothing). I plan to simply remove this check and don't make any promises about recognized/unrecognized calls. The main "promise" will be that we report "unrecognized" whenever we find something we don't understand. And that we report "recognized" for things where we think we know what's going on. But no consistency checks.

So in short - don't worry about this - we will get rid of that call eventually :-)

Reply:
For now I've opted for simply using the `RecordHandledPattern` whenever we mark something inside the new infra, this way we can move from kind of duplicating the old `Mark*FromReflectionCall` in the new infra to just use the `Mark*OnType(Hierarchy)` methods.